10:30:35  isort (ote_sdk)..........................................................Passed
10:30:35  isort (ote_cli|external).................................................Passed
10:30:38  black (ote_sdk|ote_cli)..................................................Failed
10:30:38  - hook id: black
10:30:38  - files were modified by this hook
10:30:38  
10:30:38  reformatted ote_cli/ote_cli/utils/hpo.py
10:30:38  All done! ✨ 🍰 ✨
10:30:38  1 file reformatted, 228 files left unchanged.
10:30:38  
10:30:38  black (rest).............................................................Passed
10:30:40  flake8...................................................................Passed
10:30:40  prettier.................................................................Failed
10:30:40  - hook id: prettier
10:30:40  - files were modified by this hook
10:30:40  
10:30:40  .github/labeler.yml
10:30:40  
10:30:46  mypy (ote_sdk)...........................................................Passed
10:30:46  mypy (ote_cli)...........................................................Passed
10:30:47  mypy (external)..........................................................Passed
10:31:09  pylint...................................................................Failed
10:31:09  - hook id: pylint
10:31:09  - exit code: 12
10:31:09  
10:31:09  ************* Module ote_cli.utils.hpo
10:31:09  ote_cli/ote_cli/utils/hpo.py:98:0: R0914: Too many local variables (18/15) (too-many-locals)
10:31:09  ote_cli/ote_cli/utils/hpo.py:156:0: R1721: Unnecessary use of a comprehension (unnecessary-comprehension)
10:31:09  ote_cli/ote_cli/utils/hpo.py:159:57: W0108: Lambda may not be necessary (unnecessary-lambda)

--- a/ote_cli/ote_cli/utils/hpo.py
+++ b/ote_cli/ote_cli/utils/hpo.py
@@ -50,7 +50,8 @@ def check_hpopt_available():
 def run_hpo(args, environment, dataset, task_type):
     """Update the environment with better hyper-parameters found by HPO"""
     if check_hpopt_available():
-        if task_type not in {TaskType.CLASSIFICATION, TaskType.DETECTION}:
+        if task_type not in {TaskType.CLASSIFICATION, TaskType.DETECTION,
+                             TaskType.SEGMENTATION}:
             print(
                 "Currently supported task types are classification and detection."
                 f"{task_type} is not supported yet."
@@ -135,8 +136,25 @@ def run_hpo_trainer(
     # set epoch
     if task_type == TaskType.CLASSIFICATION:
         (hyper_parameters.learning_parameters.max_num_epochs) = hp_config["iterations"]
-    elif task_type in (TaskType.DETECTION, TaskType.SEGMENTATION):
+    elif task_type == TaskType.DETECTION:
         hyper_parameters.learning_parameters.num_iters = hp_config["iterations"]
+    elif task_type == TaskType.SEGMENTATION:
+        eph_comp = [
+            hyper_parameters.learning_parameters.learning_rate_fixed_iters,
+            hyper_parameters.learning_parameters.learning_rate_warmup_iters,
+            hyper_parameters.learning_parameters.num_iters
+        ]
+
+        total_eph = sum(eph_comp)
+        eph_comp = list(map(lambda x: x * hp_config["iterations"] / total_eph, eph_comp))
+        s_ord = sorted([i for i in range(len(eph_comp))], key=lambda k: eph_comp[k], reverse=True)
+
+        for i in range(hp_config["iterations"] - sum(map(lambda x: int(x), eph_comp))):
+            eph_comp[s_ord[i]] += 1
+
+        hyper_parameters.learning_parameters.learning_rate_fixed_iters = int(eph_comp[0])
+        hyper_parameters.learning_parameters.learning_rate_warmup_iters = int(eph_comp[1])
+        hyper_parameters.learning_parameters.num_iters = int(eph_comp[2])
 
     # set hyper-parameters and print them
     HpoManager.set_hyperparameter(hyper_parameters, hp_config["params"])

--- a/ote_cli/ote_cli/utils/hpo.py
+++ b/ote_cli/ote_cli/utils/hpo.py
@@ -50,8 +50,11 @@ def check_hpopt_available():
 def run_hpo(args, environment, dataset, task_type):
     """Update the environment with better hyper-parameters found by HPO"""
     if check_hpopt_available():
-        if task_type not in {TaskType.CLASSIFICATION, TaskType.DETECTION,
-                             TaskType.SEGMENTATION}:
+        if task_type not in {
+            TaskType.CLASSIFICATION,
+            TaskType.DETECTION,
+            TaskType.SEGMENTATION,
+        }:
             print(
                 "Currently supported task types are classification and detection."
                 f"{task_type} is not supported yet."
@@ -142,18 +145,26 @@ def run_hpo_trainer(
         eph_comp = [
             hyper_parameters.learning_parameters.learning_rate_fixed_iters,
             hyper_parameters.learning_parameters.learning_rate_warmup_iters,
-            hyper_parameters.learning_parameters.num_iters
+            hyper_parameters.learning_parameters.num_iters,
         ]
 
-        total_eph = sum(eph_comp)
-        eph_comp = list(map(lambda x: x * hp_config["iterations"] / total_eph, eph_comp))
-        s_ord = sorted([i for i in range(len(eph_comp))], key=lambda k: eph_comp[k], reverse=True)
+        eph_comp = list(
+            map(lambda x: x * hp_config["iterations"] / sum(eph_comp), eph_comp)
+        )
 
-        for i in range(hp_config["iterations"] - sum(map(lambda x: int(x), eph_comp))):
-            eph_comp[s_ord[i]] += 1
+        for val in sorted(
+            list(range(len(eph_comp))),
+            key=lambda k: eph_comp[k] - int(eph_comp[k]),
+            reverse=True,
+        )[: hp_config["iterations"] - sum(map(int, eph_comp))]:
+            eph_comp[val] += 1
 
-        hyper_parameters.learning_parameters.learning_rate_fixed_iters = int(eph_comp[0])
-        hyper_parameters.learning_parameters.learning_rate_warmup_iters = int(eph_comp[1])
+        hyper_parameters.learning_parameters.learning_rate_fixed_iters = int(
+            eph_comp[0]
+        )
+        hyper_parameters.learning_parameters.learning_rate_warmup_iters = int(
+            eph_comp[1]
+        )
         hyper_parameters.learning_parameters.num_iters = int(eph_comp[2])
 
     # set hyper-parameters and print them
@@ -648,7 +659,7 @@ class HpoUnpickler(pickle.Unpickler):
 def main():
     """Run run_hpo_trainer with a pickle file"""
     hp_config = None
-    sys.path[0] = "" # to prevent importing nncf from this directory
+    sys.path[0] = ""  # to prevent importing nncf from this directory
 
     try:
         with open(sys.argv[1], "rb") as pfile:

--- a/ote_cli/ote_cli/utils/hpo.py
+++ b/ote_cli/ote_cli/utils/hpo.py
@@ -648,6 +648,7 @@ class HpoUnpickler(pickle.Unpickler):
 def main():
     """Run run_hpo_trainer with a pickle file"""
     hp_config = None
+    sys.path[0] = "" # to prevent importing nncf from this directory
 
     try:
         with open(sys.argv[1], "rb") as pfile:

--- a/tests/ote_cli/test_ote_cli_tools_segmentation.py
+++ b/tests/ote_cli/test_ote_cli_tools_segmentation.py
@@ -32,6 +32,7 @@ from common import (
     ote_eval_deployment_testing,
     ote_eval_openvino_testing,
     ote_eval_testing,
+    ote_hpo_testing,
     ote_train_testing,
     ote_export_testing,
     pot_optimize_testing,
@@ -117,6 +118,11 @@ class TestToolsSegmentation:
     @pytest.mark.parametrize("template", templates, ids=templates_ids)
     def test_ote_demo_deployment(self, template):
         ote_demo_deployment_testing(template, root, ote_dir, args)
+
+    @e2e_pytest_component
+    @pytest.mark.parametrize("template", templates, ids=templates_ids)
+    def test_ote_hpo(self, template):
+        ote_hpo_testing(template, root, ote_dir, args)
 
     @e2e_pytest_component
     @pytest.mark.parametrize("template", templates, ids=templates_ids)

